swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/notifications/NotificationQueue.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.notifications
+
+/**
+ * A queue for storing notification tasks
+ */
+interface NotificationQueue {
+  fun add(notificationTask: NotificationSender.NotificationTask)
```

Comment:
I like it when you add an import for the class, so that it shows up as `notificationTask: NotificationTask`. `option + click` on `NotificationSender` in intellij should bring up that option. It's a little cleaner to read.

Reply:
if it were a top level class yes. I prefer closer to a fully qualified name for internal classes.

Reply:
I think it's weird that `NotificationTask` is an internal class and yet it's referenced by two other files. Maybe best to move to its own file?